src/block_cache/block_chunkcache.c

```diff
@@ -0,0 +1,564 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *  All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#ifdef HAVE_DIAGNOSTIC
+#define WT_BLOCK_BEGINS_IN_CHUNK(chunk_off, block_off, chunk_size, block_size)   \
+    (block_off >= chunk_off && (block_off < chunk_off + (wt_off_t)chunk_size) && \
+      (block_off + (wt_off_t)block_size) > (chunk_off + (wt_off_t)chunk_size))
+
+#define WT_BLOCK_ENDS_IN_CHUNK(chunk_off, block_off, chunk_size, block_size) \
+    (block_off < chunk_off &&                                                \
+      (block_off + (wt_off_t)block_size) <= (chunk_off + (wt_off_t)chunk_size))
+
+#define WT_BLOCK_MIDDLE_IN_CHUNK(chunk_off, block_off, chunk_size, block_size) \
+    ((block_off < chunk_off) &&                                                \
+      (block_off + (wt_off_t)block_size) > (chunk_off + (wt_off_t)chunk_size))
+
+#define WT_BLOCK_IN_CHUNK(chunk_off, block_off, chunk_size, block_size)          \
+    (block_off >= chunk_off && (block_off < chunk_off + (wt_off_t)chunk_size) && \
+      (block_off + (wt_off_t)block_size) <= (chunk_off + (wt_off_t)chunk_size))
+
+#define WT_BLOCK_PART_IN_CHUNK(chunk_off, block_off, chunk_size, block_size)    \
+    WT_BLOCK_BEGINS_IN_CHUNK(chunk_off, block_off, chunk_size, block_size) ||   \
+      WT_BLOCK_ENDS_IN_CHUNK(chunk_off, block_off, chunk_size, block_size) ||   \
+      WT_BLOCK_MIDDLE_IN_CHUNK(chunk_off, block_off, chunk_size, block_size) || \
+      WT_BLOCK_IN_CHUNK(chunk_off, block_off, chunk_size, block_size)
+
+#define WT_CHUNK_MARK_VALID(session, chunkcache, chunk)                            \
+    if (!chunk->valid) {                                                           \
+        (void)__wt_atomic_addv32(&chunk->valid, 1);                                \
+        __wt_spin_lock(session, &chunkcache->chunkcache_lru_lock);                 \
+        TAILQ_INSERT_HEAD(&chunkcache->chunkcache_lru_list, chunk, next_lru_item); \
+        __wt_spin_unlock(session, &chunkcache->chunkcache_lru_lock);               \
+    }
+#endif
+
+#define WT_CHUNK_OFFSET(chunkcache, offset) \
+    (wt_off_t)(((size_t)offset / chunkcache->chunk_size) * chunkcache->chunk_size)
+
+#define WT_BUCKET_CHUNKS(chunkcache, bucket_id) &(chunkcache->hashtable[bucket_id].colliding_chunks)
+
+/*
+ * __chunkcache_alloc_space --
+ *     Allocate memory for the chunk in the cache.
+ */
+static int
+__chunkcache_alloc_space(WT_SESSION_IMPL *session, WT_CHUNKCACHE_CHUNK *chunk)
```

Comment:
Minor naming: I suggest just ending with `_alloc`. That is what we call all the WT allocator functions.

Reply:
Done!